source/cppfront.cpp

```diff
@@ -3065,6 +3066,7 @@ class cppfront
                 if (
                     flag_safe_subscripts
                     && i->op->type() == lexeme::LeftBracket
+                    && std::ssize(i->expr_list->expressions) == 1
```

Comment:
Maybe add a TODO, that this needs to be extended for multiple dimensions.

Reply:
Thanks! I think it's fine to do only single-dimensional bounds checks for the time being -- it will be quite a while before multidimensional is actually usable/used in the field.